frame/transaction-payment/src/lib.rs

```diff
@@ -467,6 +467,20 @@ impl<T: Trait + Send + Sync> ChargeTransactionPayment<T> where
 			Err(_) => Err(InvalidTransaction::Payment.into()),
 		}
 	}
+
+	/// Get an appropriate priority for a transaction with the given length and info.
+	///
+	/// This will try and optimise the `fee/weight` `fee/length`, whichever is consuming more of the
+	/// maximum corresponding limit.
+	///
+	/// For example, if a transaction consumed 1/4th of the block length and half of the weight, its
+	/// final priority is `fee * 2`.
```

Comment:
I'd extend the docs here by giving one more example:
```suggestion
	/// final priority is `fee * min(2, 4) = fee * 2`. If it consumed `1/4th` of the block length and the entire block weight `(1/1)`, its priority is `fee * min(1, 4) = fee * 1`. This means that the transaction which consumes more resources (either length or weight) with the same `fee` ends up having lower priority.
```